libs/linker/billsToOperation/operationsFilters.js

```diff
@@ -0,0 +1,113 @@
+const log = require('../../logger').namespace('operationsFilters')
+var isWithinRange = require('date-fns/is_within_range')
+var differenceInHours = require('date-fns/difference_in_hours')
+var { getIdentifiers, getDateRange, getAmountRange, getBillDate, getBillAmount } = require('./getterHelper')
+
+export const filterByIdentifiers = (operations, identifiers) => {
+  if (!Array.isArray(operations)) {
+    throw new Error(
+      'filterByIdentifiers cannot be called without "operations" array.'
+    )
+  }
+  if (!Array.isArray(identifiers)) {
+    throw new Error(
+      'filterByIdentifiers cannot be called without "identifiers" array.'
+    )
+  }
+
+  return operations.filter(operation => {
+    let hasIdentifier = false
+    for (const identifier of identifiers) {
+      if (operation.label.toLowerCase().indexOf(identifier) >= 0) {
+        hasIdentifier = true
+      }
+    }
+    return hasIdentifier
+  })
+}
+
+export const filterByDates = (operations, startDate, endDate) => {
+  if (!Array.isArray(operations)) {
+    throw new Error(
+      'filterByDates cannot be called without "operations" array.'
+    )
+  }
+  if (!(startDate instanceof Date)) {
+    log('warn', `typeof startDate: ${typeof startDate} = ${startDate}`)
+    throw new Error(
+      'filterByDates cannot be called without valid "startDate" date.'
+    )
+  }
+  if (!(endDate instanceof Date)) {
+    log('warn', `typeof endDate: ${typeof endDate} = ${endDate}`)
+    throw new Error(
+      'filterByDates cannot be called without valid "endDate" date.'
+    )
+  }
```

Comment:
Je trouve que ca pollue le code de la fonction tout ce code defensif. Des asserts spécialisés me semble mieux : 

```
assert(isDate(startDate), `startDate should be a date instead of ${typeof startDate}`) 
assert(isDate(endDate), `endDate should be a date instead of ${typeof endDate}`) 
```